### ISSUE TITLE
Deadly Poison - Combat + Sub

### ISSUE THREAD

**Issue by BalanCV** [gamefreedomgit/Maelstrom#3863]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**
There is some inconsistency with Deadly poison where it could not proc for 12+ sec making it to fall off while rogue is attacking target even with SnD on and casting abilities. This is observed in the sub and combat specs, might affect assa as well (whatever core issue it is) but it is probably masked by the increased proc percentage of both Improved poisons and envenom.
I believe a fast attack class sticked to the target should not experience poison falloff normally.

Note: After I recorded the video below I made an unrealistic combat rogue with 27% hit rating and 47 expertise. I smacked the target dummy for 15+ minutes and have actually experienced 1 falloff on that as well and a few "almost-falloffs" (same test scenario: keep SnD up, sinister when building up combo for SnD)
This makes me pretty certain that there is some underlying issue, either simply with proc frequency or something else causing inconsistency that should be investigated.

I cannot record such a long test video as my test above but if you have the patience to do it yourself, it will happen eventually.

**How to reproduce:**
https://youtu.be/dokZ8Qw7r20

Noticeable at the very start of the video, it almost falls of at random points as well 2:30.
Due to inconsistency of the issue, you can test personally for a long period of time and even a 5stack with SnD on and casting abilities can still fall off, but I cannot record for that long period of time.

**How it should work:**

**Database links:**

**Comment by bonfax4** [comment]
It did happen on retail but it was pure rng and uncommon. There used to be [discussions](https://www.mmo-champion.com/threads/931878-Combat-and-shiv-question?highlight=Combat+rogue+shiv) to understand if it was worth shiving (force deadly refresh ) in these situations for combat. Sub skips spellhit  and  combat doesn't prioritize it over haste.The dps gain for shiving as combat was so minor that it was not worth worrying about it. Assass has envenom/spell hitcap and an off hand weapon attack.

**Comment by BalanCV** [comment]
https://www.youtube.com/watch?v=1TNIy62oE24&ab_channel=Pheron12

I was watching this video and found it interesting that his deadly didn't ever go halfway before refreshing (think he didn't shiv once during the fight). To be fair this isn't the best comparison as this guy has both melee haste raid buff and legendary daggers which I didn't use in my test, but I guess its informative in some way. However, even if falloff did happen in retail I'm guessing it was so rare that most people didn't even notice. Here it seems to happen more often, at least according to my friend who raids with the spec and cries about it every few minutes :))

I remember very early in development there was some changelog about rogue poisons proccing too much, it might be unrelated but it could be at that time they shifted proc chance lower than it should and it could just be an easy fix of just upping the proc chance a lil' bit. Regardless, just speculation on my part, the dev will know best once he looks through the data.

I actually posted this related to your earlier post as another reason why combat/sub might be underperforming currently. Wanted to just lazy comment, but things posted in comments seem not likely to be addressed.

**Comment by bonfax4** [comment]
As i said it'e a small chance but its still a chance. Is deadly dropping for you every single fight ?

**Comment by bonfax4** [comment]
I'll run a swing test to see if deadly has its 30% proc chance

**Comment by BalanCV** [comment]
Yes, I did a build on Test Realm with 27% hit to bypass any offhand miss and even with that I managed to have it fall off once and almost fall off a few times in a 15min session.
Also just logged in on normal realm 2.5min on heroic dummy:
37 backstabs + 103 melee + 80 glancing melee = 220 hits (didn't add misses)
31 deadly (1 apply + 30 refresh)
31/220 = 14% proc chance

Surprisingly if you factor out glancing melees and ability hits:
31/103= 30% proc chance
Could this be the issue? Maybe its not factoring in abilities and glancing?

**Comment by bonfax4** [comment]
You'll prob need a bigger sample pool for accuracy but yh it sounds off. Glanced blows are still hits. Do a 1k+ swing test, the more the better.

**Comment by bonfax4** [comment]
Uff if this issue carriers over to every on hit proc, Enhancement shamans must be getting shafted pretty hard in pve with all their on hit procs. ( Maelstrom weaps, windfury, flametongue )

**Comment by BalanCV** [comment]
Conclusive tests (ignore ones above rushed forgot to add crit hits)

1k hits on Heroic Dummy (ignoring miss/dodge) - fell off once
275 normal
250 crits
245 glancing
102 deadlys (2 applications 100 refreshes)
102/(275+250+245)= 13% proc chance

1k hits on 85 dummy ( no glancing here) - didn't fall off
555 normals
273 crits
104 deadlys
104/(555+273) = 12.5% proc chance

Seems like its not glancing, just too low proc chance. This will affect all specs since its the base poison that's not 30%

**Comment by bonfax4** [comment]
Here: 324/ (866 + 166 ) = 31.3%
![d32a5f828f8b20138b7d8b31b9447b98](https://github.com/gamefreedomgit/Maelstrom/assets/16911730/d5d4c6e4-0416-49bf-9009-bfe660cebf84) ![49faa29630e4fbb76458356ddc290b94](https://github.com/gamefreedomgit/Maelstrom/assets/16911730/d1a42da7-12c5-4d28-9480-71ef09484807) ![f119fe3e4e91139bf897857074aeb20e](https://github.com/gamefreedomgit/Maelstrom/assets/16911730/8b40435f-72be-4377-8fae-3b3fc2b677fc)

**Comment by BalanCV** [comment]
Yep, I'm getting closer to 28%. That also brings us back to square one regarding the drop-off.

**Comment by bonfax4** [comment]
Honestly feel like this is just normal. Will get better as you gear because of higher haste levels. Other poisons are ppm based btw so i'm sure deadly main hand / instant off ends up being a dps loss because of weapon speed.
